### ISSUE TITLE
rogue Uniprot ID in physical interactions data output file?

### ISSUE THREAD

**Issue by ValWood** [pombase/pombase-chado#532]
Dan and Giorgio have reported a rogue Uniprot ID in  this file:

In the file
 http://curation.pombase.org/dumps/releases/pombase-chado-latest/exports/pombase-go-physical-interactions.tsv.gz

SPBC409.05	UniProtKB:O59671	PMID:15147268

any ideas where it comes from? I'll have a poke around....Very odd.....

Val

**Comment by ValWood** [comment]
I checked in artemis as that's the only way it could have got into the manual annotation, its not there (could not be via canto). 

I grepped for the  Uniprot ID in the current GO GAF  on the assumption that if all of the interaction data is generated form GO, if this exists it should be in this file. It isn't there.

I looked at the UniProt annotation to see if it was one of theirs that we were importing which had not had the "with" field converted to a PomBase systematic ID but I can't see any such interactions in UniProt for this SPBC409.05

Stumped!

**Comment by ValWood** [comment]
Can you stick this  to the top of the pile?

**Comment by kimrutherford** [comment]
We are getting this annotation from GOA:

UniProtKB   Q9Y709  skp1            GO:0005515      PMID:15147268   IPI     UniProtKB:O59671        F       Suppressor of kinetochore protein 1     SKP1_SCHPO|skp1|psh1|sph1|SPBC409.05    protein taxon:284812    20070103        UniProt

It's on this page:
  https://www.ebi.ac.uk/QuickGO/GProtein?ac=Q9Y709

Does that explain things?

**Comment by ValWood** [comment]
Yes and no ;)

On this page,
https://www.ebi.ac.uk/QuickGO/GProtein?ac=Q9Y709
 there are lots of annotations of this type, but they do not end up with a UniProt ID in the output file. The ID at some point the thing in the "with" field should be converted to a pombase identifier?

There are other issues with duplication, I'll put that in a different ticket. The duplication might be part of the cause. In reality some of these annotations are triplicated in GOA. They are duplicated on our gene pages. We need to improve the filtering here so we only end up with each annotation once.

It is possible (although probably unlikely), that the reason we don't see this more than once in the file for EsyN is due to a weird combination of source/filtering getting rid of the other  ones like this. I can;t remember exactly how this works.

**Comment by kimrutherford** [comment]
> The ID at some point the thing in the "with" field should be converted to a pombase identifier?

We don't do any conversion when loading.  I'll dig deeper when I'm back.

**Comment by ValWood** [comment]
No hurry I told Dan we'd let him know when ready.
I'll also have a look. I suspect that filtering removes most as duplicates and so the vast majority disappear. I can double check this first.
Already looked at the redundancy in these annotations so I have the info somewhere....

**Comment by ValWood** [comment]
Its odd. The one in the file is describing skp1 biding to pof4. However there are other annotations which will have UniPRot IDs in the with field from this paper
See this ticket
https://github.com/pombase/curation/issues/859
Some interactions are curated by PomBAse unitpto and intact. Ideally you would filter the intace and the Uniprot ones. This might partly be happening, but on the skp1 page there are definitly some duplicates (but never any triplicates, even when all 3 , pombase/uniprot & intact) have made the annotations.

So i'm not sure why the only one which shows up with a uniprot ID in the with filed is this  particylar one?

There are others which we did not make at PomBase. But, I think  for these we might have made the reciprocal annotations. I think this is another reason why we want to do this:
https://github.com/pombase/pombase-chado/issues/433
Before any filtering the PomBase annotations need to be 'reciprocated' so that when imported annotations are filtered it is not dependant on the  directionality of  the curation.

Let me know if you need to chat about this. I think the first thing is for you to tell me what you think happens currently (I can't remember exactly)

val

**Comment by ValWood** [comment]
Just so I can get this off my To Do list and onto a tracker

Note to self 
"easyN  check updated, test metaphase anaphase transition chromosome segregation should update to show SPBC32H8.09 and gtb1"

Assign this ticket to me when this task is done.

Thanks

Val

**Comment by kimrutherford** [comment]
> We don't do any conversion when loading.

Sorry for the confusion, I was wrong about that.  I forgot that we map UniProt ID to pombe IDs using this file:
  ftp://ftp.ebi.ac.uk/pub/databases/pombase/pombe/Mappings/PomBase2UniProt.tsv

It's downloaded for each Chado build.  That stray ID (O59671) isn't in the mapping.

**Comment by ValWood** [comment]
Thanks. Mapping file issues are in progress. 

Hang fire, I'll let you know when it's sorted.

**Comment by ValWood** [comment]
We are waiting on
https://www.ebi.ac.uk/panda/jira/browse/PB-2408
but in the meantime maybe a check and report in logs if an entry  isn't mapping to UniProt correctly?

**Comment by kimrutherford** [comment]
> but in the meantime maybe a check and report in logs if an entry isn't mapping to UniProt correctly?

I've added a warning instead of exporting the interaction with a UniProt ID.

**Comment by kimrutherford** [comment]
> Hang fire, I'll let you know when it's sorted.

Is there more to do before I close this issue?

**Comment by ValWood** [comment]
I need to open a new ticket about how we handle duplicate  annotations from uniprot generally. As this won't be needed until you are on to the filtering tasks, I wont get to it immediately.

Could you let me know when there is an update which picks up this change. I'll let Giorigio and Dan know to update EsyN

Thanks

**Comment by kimrutherford** [comment]
> Could you let me know when there is an update which picks up this change.

I wasn't planning to do any new loads for me because updating the ontologies is still a pain.  I can go ahead if needed.  It's probably quicker just to edit the file and remove the line that has a UniProt ID if that is holding up Dan and Giorgio.

**Comment by ValWood** [comment]
We'll not worry about this until I get back. 

Dan and Giorgio have both left the department now, they are babysitting this, but they don't want to do anything manually.
I want to check that everything is working as part of our release pipeline  so the update can wait until I can check it all properly.

There are still issues with the Uniprot file mapping.
https://www.ebi.ac.uk/panda/jira/browse/PB-2408

If we do have a chado update,  and the corrected file is in place I'll let them know, but we won't do anything as a temporary fix because they'll get annoyed if anything goes wrong.

**Comment by kimrutherford** [comment]
> If we do have a chado update, and the corrected file is in place I'll let them know

The file from the new load looks good - no UniProt IDs.

The links they use will be in place after we make a release.

**Comment by ValWood** [comment]
These files look fine now. There is a blank line at the bottom of both files
pombase-go-substrates.tsv
and
pombase-go-physical-interactions.tsv

should you remove that?
Val

**Comment by kimrutherford** [comment]
I can't see a blank line at the end of either file.

**Comment by ValWood** [comment]
Seems to be, from here?
https://curation.pombase.org/dumps/releases/pombase-chado-v60-2016-05-12/exports/
I checked again

**Comment by kimrutherford** [comment]
I've checked all the files there.  All have a new line at the end of each line and the last line isn't empty.  How are you viewing the files?

**Comment by ValWood** [comment]
I'm downloading and opening in textedit. When I scroll down I see a blank line?

**Comment by ValWood** [comment]
Double checking but files are currently empty.

**Comment by kimrutherford** [comment]
>  When I scroll down I see a blank line?

In most editors if the last line of the file has a newline at the end, it shows a blank line at the end of file.  There are the same number of non-empty lines as there are newline characters.  If it doubt have a look at the file with "less" on the command line and then press capital G to go to end of file.

**Comment by ValWood** [comment]
OK I never noticed that before!
